bootloader/prebootloader/src/rtl872x/part1/rtl_support.c

```diff
@@ -223,7 +223,7 @@ static uint32_t boot_reason_set(void) {
     /* backup boot reason to REG_LP_SYSTEM_CFG2 */
     temp_bootcfg = HAL_READ32(SYSTEM_CTRL_BASE_LP, REG_LP_SYSTEM_CFG2);
     temp_bootcfg |= tmp_reason;
-    HAL_WRITE32(SYSTEM_CTRL_BASE_LP, REG_LP_SYSTEM_CFG2, tmp_reason);
+    HAL_WRITE32(SYSTEM_CTRL_BASE_LP, REG_LP_SYSTEM_CFG2, tmp_reason); // Should tmp_reason be `temp_bootcfg`? If we have deepsleep or brownout boots, we will clear some of the bits in REG_LP_SYSTEM_CFG2
```

Comment:
This is unrelated to the PR, but I think there is a subtle bug in the RTK SDK, where the `REG_LP_SYSTEM_CFG2` might be getting written incorrectly on boot in some circumstances??

Reply:
Good question.  @avtolstoy ?

Reply:
I think the only stuff we'd lose is some ROM-specific params, the boot reason should be accurate irrespective of that as it's collected from AON/SYSON registers into `tmp_reason`.